### ISSUE TITLE
Structured reference to table in chart

### ISSUE THREAD

**Issue by pmqs** [jmcnamara/excel-writer-xlsx#234]
I attempting to create a chart that gets its data from an excel table. If I give the table data to the `add_series` method using explicit cell references all is fine. 

If I use  the structured reference syntax `table[column]`  in  the `add_chart` method I'm getting mixed results.

So something like this works, where `categories` uses a structured reference to a column `Product`, but `values` uses an explicit cell reference.

```perl
$chart1->add_series(
    name       => 'MyChart',
    categories => '=MyTable[Product]', 
    values     => '=Sheet1!$C$5:$C$8',
);
```

but attempting to use a structured reference with `values` and the column `Quarter1` fails

```
$chart1->add_series(
    name       => 'MyChart',
    categories => '=MyTable[Product]', 

    # This doesn't work
    values     => '=MyTable[Quarter1]',
);
```

The error I get is

> Can't use an undefined value as an ARRAY reference at /home/perl/install/lib/site_perl/5.24.0/Excel/Writer/XLSX/Chart.pm line 5297.

Here is the full test script I'm using
```perl
#!/usr/bin/perl

use strict;
use warnings;
use Excel::Writer::XLSX;

my $workbook    = Excel::Writer::XLSX->new( "table.xlsx" );

my $worksheet  = $workbook->add_worksheet();

# Some sample data for the table.
my $data = [
    [ 'Apples',  10000, 5000, 8000, 6000 ],
    [ 'Pears',   2000,  3000, 4000, 5000 ],
    [ 'Bananas', 6000,  6000, 6500, 6000 ],
    [ 'Oranges', 500,   300,  200,  700 ],

];

# Add a table to the worksheet.
$worksheet->add_table( 'B4:F8', { 
              name => "MyTable",
             columns => [
            { header => 'Product' },
            { header => 'Quarter1' },
            { header => 'Quarter 2' },
            { header => 'Quarter 3' },
            { header => 'Quarter 4' },
        ] } );

# Table data can also be written separately, as an array or individual cells.
$worksheet->write_col( 'B5', $data );

# Create a new chart object. In this case an embedded chart.
my $chart1 = $workbook->add_chart( type => 'bar', embedded => 1 );

# Configure the first series.
$chart1->add_series(
    name       => 'MyChart',
    categories => '=MyTable[Product]', 

    # This doesn't work
    values     => '=MyTable[Quarter1]',

    # This works
    # values     => '=Sheet1!$C$5:$C$8',
);

# Set an Excel chart style. Blue colors with white outline and shadow.
$chart1->set_style( 11 );

# Insert the chart into the worksheet (with an offset).
$worksheet->insert_chart( 'H4', $chart1 );

$workbook->close();

```

Version Info below.
```
Perl version   : 5.024000
OS name        : linux
Module versions: (not all are required)
                 Excel::Writer::XLSX     	1.00
                 Spreadsheet::WriteExcel 	(not installed)
                 Archive::Zip            	1.57
                 XML::Writer             	(not installed)
                 IO::File                	1.16
                 File::Temp              	0.2304

```

**Comment by jmcnamara** [comment]
Thanks for the detailed bug report. That's a bug.

The reason this happens is that Excel caches/duplicate cell worksheet data in the chart. For consistency with Excel the Excel::Writer::XLSX reads the user supplied cell data for ranges like `Sheet1!A1:A5` and add it to the chart. This isn't strictly required since Excel recalculates the data anyway. Excel::Writer::XLSX doesn't try to parse complex ranges and in this case it shouldn't try to parse the table reference.

As a workaround you can supply an empty arrayref of category and value data which will prevent the range. Like this:

```perl
$chart1->add_series(
    name            => 'MyChart',
    categories      => '=MyTable[Product]',
    values          => '=MyTable[Quarter1]',
    categories_data => [],
    values_data     => [],
);
```

Use this for now and I'll add a proper fix in time.

**Comment by pmqs** [comment]
Thanks for the swift response. The workaround fixes the issue.

**Comment by jmcnamara** [comment]
This is fixed in version 1.01 which is on its way to the CPAN mirrors.
